lib/services/service-scaffold.ts

```diff
@@ -252,7 +252,9 @@ export abstract class ServiceScaffold<T> extends WorkerClient<T> implements Serv
 	 * Get the service name, as required by the framework.
 	 * @return  Name of the service.
 	 */
-	abstract get serviceName(): string
+	get serviceName(): string {
+		return this.serviceName;
+	}
```

Comment:
1) Creates an infinite loop..
2) Does not insist that the variable is defined.
3) Would be nice to remove overrides of this that follow the pattern.

Reply:
Something like https://github.com/resin-io-modules/resin-procbots/commit/c71844cd75ec59b6e4f43bfc9617b4413cdecb3d

Reply:
@sqweelygig Actually, having now had a closer look at this, I really don't like two things:

1. The reason this was changed is because the ServiceScaffold constuctor was trying to use `serviceName`, which is strictly prohibited (this is an abstract property, and therefore invalid)
2. The better way to do this is to pass `serviceName` into the `ServiceScaffold` constructor if we want to do this, and explicitly set it there so it can be used and called by the implementer, removing the private instances from each Service.

I do not have anymore time to do 2. I'm backing out the change I made, and commenting out the logging in the `ServiceScaffold` constructor. If you'd like to make the relevant changes on this PR, that would be helpful.